inc/trilogy/protocol.h

```diff
@@ -597,6 +604,128 @@ typedef struct {
     size_t data_len;
 } trilogy_value_t;
 
+typedef struct {
```

Comment:
Would it be useful to reference https://dev.mysql.com/doc/internals/en/binary-protocol-value.html here to explain where this comes from?

Reply:
Also might be good to annotation each part in the `union` then with which binary protocal value type(s) it implements?

Reply:
Annotating the union is definitely a good idea 👍  It's on my internal list of things to do before taking this out of draft, I'll add it to the pull request body checklist so I don't forget ;)

In general, another documentation pass wouldn't hurt.